adafruit_avrprog.py

```diff
@@ -132,10 +156,8 @@ def program_file(self, chip, file_name, verbose=False, verify=True):
         self.begin(clock=clock_speed)
 
         # create a file state dictionary
-        file_state = {"line": 0, "ext_addr": 0, "eof": False}
-        with open(file_name, "r") as file_state[  # pylint: disable=unspecified-encoding
-            "f"
-        ]:
+        file_state = {"line": 0, "ext_addr": 0, "eof": False, "f": TextIOWrapper}
```

Comment:
Since it's going to get replaced, you could just make the `"f"` key equal to `None` and not import `TextIOWrapper`.

Reply:
I did move FileState into a TypedDict and change the initial value to None. But also kept TextIOWrapper imported and set the type of `f` in the TypedDict to `Optional[TextIOWrapper]` so that it shows the possible values of None or an open file. 

If there is a better way to do that we can make further changes in a follow up PR.